### DISCUSSION TITLE
Ability to suppress specific metrics

### DISCUSSION THREAD

**Discussion by shayyxi** [autometrics-dev/autometrics-shared#41] [Ideas]
I came to know about this cool instrumentation framework while visiting your Kubecon booth. While visiting some other booths, there were some tools focused on controlling the metrics quantity reaching the datastore for data cost-saving purposes. Which basically gave me this idea I am going to share here. I was thinking about the ability in this framework to suppress some specific metrics for a function. In this way, if a developer is not interested in some specific metrics for some specific function, they can suppress those specific metrics for that function. 

What do you guys think about this?

**Comment by emschwartz** [comment]
Hi @shayyxi thanks for the suggestion!

We definitely could add an option for this, though I'd be curious about the exact use case for it.

Right now, we're only tracking [2 metrics per function](https://github.com/autometrics-dev/autometrics-shared/blob/main/SPEC.md#metrics) by default. The most expensive is the histogram for tracking latency, but I'd probably want to hear a specific use case for skipping that before adding more options to the library.
